packages/smooth_app/lib/services/smooth_service.dart

```diff
@@ -0,0 +1,28 @@
+/// Generic interface for a service (eg: logger, analytics…) containing
+/// one or multiple implementations
+abstract class SmoothService<T extends SmoothServiceImpl> {
```

Comment:
If you ask me, not the best idea to have 2 files with almost the same name and two classes with almost the same name, like `smooth_service.dart` and `smooth_services.dart`.

Reply:
What would you do instead?

Reply:
I would NOT have class and files with similar names.
For instance, what about `AbstractSmoothService` and `SmoothServiceRepository`?
To be honest I'm not a big fan of the `Impl` suffix either (we can afford names with real words) but at least with `Impl` I expect an `impl`(ementation) of an abstract class/interface, therefore the name is descriptive enough.